sdk/test-utils/perf/test/event/mockEventReceiver.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { generateUuid } from "@azure/core-http";
+
+export interface Event {
+  body: string;
+}
+
+export interface SubscribeOptions {
+  /**
+   * The maximum number of concurrent calls that the library
+   * can make to the user's event handler.
+   * - **Default**: `1`.
+   */
+  maxConcurrentCalls?: number;
+  /**
+   * Raises the error after the specified time.
+   */
+  raiseErrorAfterInSeconds: number;
+}
+
+/**
+ * The general event handler interface (used for streaming events).
+ */
+export interface EventHandlers {
+  /**
+   * Handler that processes event.
+   */
+  processEvent(event: Event): Promise<void>;
+  /**
+   * Handler that processes errors that occur during receiving.
+   */
+  processError(err: Error): Promise<void>;
+}
+
+export class MockEventReceiver {
```

Comment:
I think this mock test was designed to be similar to `ServiceBusReceiver`, which is slightly different than the `MockEventProcessor` in .NET, which is designed to be similar to `EventProcessorClient`.

If you could port the `MockEventProcessor` and `MockEventProcessorEventTest` from .NET to JS, we could use these tests to compare the results across languages.  Otherwise I think the JS `MockEventReceiver` and .NET `MockEventProcessor` are too different to directly compare.  For example, `MockEventProcessor` supports a concept of "partitions", while `MockEventReceiver` has the concept of `maxConcurrentCalls`, which are similar but not directly comparable.

https://github.com/Azure/azure-sdk-for-net/tree/main/common/Perf/Azure.Sample.Perf/Event

Reply:
maxConcurrentCalls is a concept of service bus receiver, it is not the equivalent of partitions I believe.
We do have partitions for event-hubs receiver.

Reply:
Added MockEventReceiver2 that is closer to the event-hubs receiver.